index.d.ts

```diff
@@ -57,17 +57,17 @@ declare namespace Router {
     store: any
   ) => void;
 
-  interface ConstraintStrategy<V extends HTTPVersion> {
+  interface ConstraintStrategy<V extends HTTPVersion, ConstraintVersion = string> {
```

Comment:
I think it might make more sense to call this property `ConstraintValue` or something like that. I know it says `version` in all the types below, but these strategies can actually be used for other constraints than version, like `host` or maybe the `Accept` header someday. I think the name of that argument in the types should be changed too.